pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
this does not seem right either. we should still report failure if reconcile fails, not just when we delete the cluster.

also for cluster delete case, the tpr might already be deleted. there is no point to update it.